backend/app/api/api_v1/endpoints/skill.py

```diff
@@ -29,12 +29,83 @@ def get_all_skill(
     return skills
 
 
-@router.get("/courses")
+@router.get("/active", response_model=List[schemas.Skill])
+def get_active_skill(
+    db: Session = Depends(deps.get_db),
+    skip: int = 0,
+) -> Any:
+    """
+    Retrieve all active skills.
+    """
+    skills = crud.skill.get_active(db, skip=skip)
+    if not skills:
+        raise HTTPException(
+            status_code=404,
+            detail="No active skills found",
+        )
+    return skills
+
+
+@router.get("/courses/active")
+def get_active_skills_and_courses(
+    db: Session = Depends(deps.get_db),
+) -> Any:
+    """
+    Get all active skills and all their courses.
+    """
+    sql_query = text(
+        """
+        SELECT
+            s.skill_id, s.skill_name, s.skill_desc, s.is_active, sc.courses
+        FROM
+            skill s
+            LEFT JOIN (
+                SELECT
+                    sc.skill_id,
+                    JSON_ARRAYAGG(JSON_OBJECT(
+                        'course_id', c.course_id,
+                        'course_name', c.course_name,
+                        'course_desc', c.course_desc,
+                        'course_status', c.course_status,
+                        'course_type', c.course_type,
+                        'course_category', c.course_category
+                    )) courses
+                FROM skill_course sc, course c
+                WHERE sc.course_id = c.course_id
+                GROUP BY sc.skill_id
+            ) sc ON s.skill_id = sc.skill_id WHERE s.is_active = 1;
+    """
+    )
+    db_cursor_obj = db.execute(sql_query)
+    if not db_cursor_obj:
+        raise HTTPException(
+            status_code=404,
+            detail="Error getting active skills with their courses",
+        )
+
+    skills_with_courses = db_cursor_obj.all()
+    if not skills_with_courses:
+        raise HTTPException(
+            status_code=404,
+            detail="No active skills with courses in the database",
+        )
+
+    skills_with_courses = [dict(r) for r in skills_with_courses]
+    for row in skills_with_courses:
+        if row["courses"] is not None:
+            row["courses"] = json.loads(row["courses"])
+        else:
+            row["courses"] = []
```

Comment:
/nit I don't think we need the "is not None" check here, since if `row["courses"]` is `None` or `[]`, we still get the same result

Reply:
the "is not None" check is the fix for the 500 error i got previously. Basically json.loads cannot be called on a NoneType.
the else block is to ensure we don't get an error when we call map() on the result at the frontend side